core/src/main/java/org/elasticsearch/index/fielddata/IndexFieldDataService.java

```diff
@@ -108,7 +110,20 @@ public synchronized void clearField(final String fieldName) {
     @SuppressWarnings("unchecked")
     public <IFD extends IndexFieldData<?>> IFD getForField(MappedFieldType fieldType) {
         final String fieldName = fieldType.name();
-        IndexFieldData.Builder builder = fieldType.fielddataBuilder();
+        IndexFieldData.Builder builder;
+        try {
+            builder = fieldType.fielddataBuilder();
+        } catch (InvalidParameterException e) {
+            MappedFieldType keywordSubField = mapperService.subfieldType(fieldType.name(), KeywordFieldMapper.KeywordFieldType.class);
+            if (keywordSubField != null) {
```

Comment:
It's also possible for `mapperService.fullName()` to return null if the specified field is not found, so I thought the same consistent approach would also work here.